lib/mysqlService.js

```diff
@@ -231,6 +221,59 @@ class MysqlService {
         }
     }
 
+
+    /**
+     * Stream the results of a query.
+     * @param {String} sql
+     * @param {ReadableOptions} [streamOptions] Options to be passed to the stream constructor.
+     * @return {stream.Readable}
+     */
+    streamQuery(sql, streamOptions = {}) {
+      return this._streamQuery(this._getReadPool(), sql, streamOptions);
+    }
+
+    /**
+     * Stream the results of a query from the master pool
+     * @param {String} sql
+     * @param {ReadableOptions} [streamOptions] Options to be passed to the stream constructor.
+     * @return {stream.Readable}
+     */
+    streamQueryFromMaster(sql, streamOptions = {}) {
+      return this._streamQuery(this._writePool, sql, streamOptions);
+    }
+
+
+    /**
+     * @param pool
+     * @param {String} sql
+     * @param  {ReadableOptions} streamOptions
+     * @return {stream.Readable}
+     */
+    _streamQuery(pool, sql, streamOptions) {
+      this._logSqlQuery(sql);
+
+      const defaults = { objectMode: true };
+      const opts = Object.assign(defaults, streamOptions);
```

Comment:
Same comment as above for setting defaults. I usually just do it at the top and overwrite the same options object: 

```javascript
streamOptions = _.defaults(streamOptions, {
    objectMode: true
});
```